airflow/models/baseoperator.py

```diff
@@ -299,8 +299,8 @@ def __init__(
         task_id: str,
         owner: str = conf.get('operators', 'DEFAULT_OWNER'),
         email: Optional[Union[str, Iterable[str]]] = None,
-        email_on_retry: bool = True,
-        email_on_failure: bool = True,
+        email_on_retry: bool = conf.getboolean('email', 'default_email_on_retry'),
```

Comment:
Do we want this globally? I think DAG level is fine enough, what is your use-case?

Reply:
The main reason for proposing this change is that I believe not every team uses emails to alert on job failure (our team only use slack alert). So every time we create a dag, we need to remember to set these 2 options to `False`, which is error-prone sometimes. Once we does forget to set to False and pass the code review, the error log would be messed with email related error when a task fail. I guess I can also have some some safe guard in CI/CD, but it will be nice if we can override the default setting globally. 

I am not sure what problem would it cause if we set it globally. Can you explain a bit about your concern?

Reply:
BTW, are you concerned about both settings or just the `email_on_retry` ?

Reply:
Don't get me wrong, I definitely don't have a serious concern over this. Just want to think it through.

So currently we can set this flag on task level, in default_args so that it applies to all tasks, dag level and with this change config.

Reply:
Can we add a test so that if someone modifies this code and changes it to `conf.get` instead of `conf.getboolean`, that test should fail.

Also let's add a fallback value too so `conf.getboolean('email', 'default_email_on_retry', fallback=True)`

Reply:
Hi @kaxil, I updated the code based on your feedback.
1. Add `fallback=True` to both email_on_retry and email_on_failure
2. Add a check in the pre-commit config which will fail when it's not `conf.getboolean` (this is my first time to write something about pre-commit using pygrep. Let me if it needs to be modified.

Also, 
- I added a similar check to the `retreis` argument i.e. fail when it's not `conf.getint`
- Resort the pre-commit check order: now all checks with id: `base-operator` goes together.

PTAL. Thanks!